src/r1cs/mod.rs

```diff
@@ -78,26 +71,34 @@ pub struct RelaxedR1CSInstance<E: Engine> {
   pub(crate) u: E::Scalar,
 }
 
+/// A type for functions that hints commitment key sizing by returning the floor of the number of required generators.
 pub type CommitmentKeyHint<E> = dyn Fn(&R1CSShape<E>) -> usize;
 
-impl<E: Engine> R1CS<E> {
```

Comment:
What was the rationale for removing the type?

Reply:
It's a token struct with only one method, which is (as Self is a token), independent of self (the [unused_self](https://rust-lang.github.io/rust-clippy/master/index.html#/unused_self?levels=allow) lint fires on it). We didn't see the justification for the wrapper's complexity.